src/treecons.ml

```diff
@@ -0,0 +1,131 @@
+(** Tree constructors: a simpler representation of terms to be used by a
+    decision tree. *)
+
+open Extra
+
+(** A constructor is the representation of a symbol along with the number of
+    arguments to which it is applied. *)
+type treecons =
+  | Symb of
+      { c_mod : string list
+      (** Module name where the symbol of the constructor is defined. *)
+      ; c_sym : string
+      (** Symbol of the constructor. *)
```

Comment:
If s is a sym, c_mod corresponds to s.sym_path and c_sym to s.sym_name. Why not replacing these 2 fields by a single field of type sym?